src/mon/FSCommands.cc

```diff
@@ -436,18 +437,38 @@ class SetHandler : public FileSystemCommandHandler
         return r;
       }
 
+      ss << fs->mds_map.get_fs_name();
+
       fsmap.modify_filesystem(
           fs->fscid,
           [is_down](std::shared_ptr<Filesystem> fs)
       {
-        if (is_down) {
-          fs->mds_map.set_flag(CEPH_MDSMAP_DOWN);
-        } else {
-          fs->mds_map.clear_flag(CEPH_MDSMAP_DOWN);
-        }
+	if (is_down) {
+	  fs->mds_map.set_flag(CEPH_MDSMAP_DOWN);
```

Comment:
Now this would be the only place where we set CEPH_MDSMAP_DOWN. It would not provide for a way to set it while leaving any MDS active.